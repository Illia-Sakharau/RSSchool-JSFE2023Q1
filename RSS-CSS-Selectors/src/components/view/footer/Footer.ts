import './footer.scss';
import htmlToElement from '../../../utils/htmlToElement';
import gitImg from '../../../assets/github.svg';
import schoolLogoImg from '../../../assets/rsslogo.svg';

export default class Footer {
  public draw(): HTMLElement {
    const template: string = `<footer>
        <a href="https://github.com/Illia-Sakharau" target="_blank" class="author">
          ${gitImg}
          <span>Illia Sakharau</span>
        </a>
        <a href="https://rs.school/js/" target="_blank" class="school">
          ${schoolLogoImg}
        </a>
        <div class="copyright">
          Â© 2023
        </div>
      </footer>
    `;

    return htmlToElement(template);
  }
}
