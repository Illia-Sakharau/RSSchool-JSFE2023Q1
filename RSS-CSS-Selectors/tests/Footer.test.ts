import Footer from '../src/components/view/footer/Footer';

jest.mock('../src/components/view/footer/footer.scss', () => ({}));
jest.mock('../src/assets/github.svg', () => 'github.svg');
jest.mock('../src/assets/rsslogo.svg', () => 'rsslogo.svg');
const footer = new Footer();

afterAll(() => {
  jest.restoreAllMocks();
});

describe('Test drow footer', () => {
  test('should return correct answer', () => {
    expect(footer.draw()).toBeDefined();
  });
  test('should return footer', () => {
    expect(footer.draw()).toMatchSnapshot();
  });
});
