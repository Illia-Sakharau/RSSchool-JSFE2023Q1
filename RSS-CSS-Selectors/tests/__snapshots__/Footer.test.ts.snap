// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test drow footer should return footer 1`] = `
<footer>
  
        
  <a
    class="author"
    href="https://github.com/Illia-Sakharau"
    target="_blank"
  >
    
          github.svg
          
    <span>
      Illia Sakharau
    </span>
    
        
  </a>
  
        
  <a
    class="school"
    href="https://rs.school/js/"
    target="_blank"
  >
    
          rsslogo.svg
        
  </a>
  
        
  <div
    class="copyright"
  >
    
          Â© 2023
        
  </div>
  
      
</footer>
`;
